docs/distribution/README.md

```diff
@@ -0,0 +1,927 @@
+If the header is not present, the client can assume that all results have been received.
+
+> __NOTE:__ In the request template above, note that the brackets are required. For example, if the url is `http://example.com/<name>/manifests/sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042/references/list?n=20&last=b`, the value of the header would be `http://example.com/<name>/manifests/sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042/references/list?n=20&last=b>; rel="next"`.
+> Please see [RFC5988](https://tools.ietf.org/html/rfc5988) for details.
+
+Compliant client implementations SHOULD always use the `Link` header value when proceeding through results linearly. The client MAY construct URLs to skip forward in the list of tags.
+
+To get the next result set, a client would issue the request as follows, using the URL encoded in the described `Link` header:
+
+```HTTP
+GET /v2/<name>/manifests/sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042/references/list?n=<n from the request>&last=<last tag value from previous response>
+```
+
+The above process should then be repeated until the `Link` header is no longer set in the response.
+
+The tag list result set is represented abstractly as a lexically sorted list, where the position in that list can be specified by the query term `last`. The entries in the response start _after_ the term specified by `last`, up to `n`
+entries.
+
+#### Google Paging API
+
+From [Google API Design Guides][google-paging-api]
+
+To support pagination (returning list results in pages) in a List method, the API shall:
+
+* define `a string` field `page_token` in the `List` method's request message. The client uses this field to request a specific page of the list results.
+* define an `int32` field `page_size` in the `List` method's request message. Clients use this field to specify the maximum number of results to be returned by the server. The server **may** further constrain the maximum number of results returned in a single page. If the `page_size` is `0`, the server will decide the number of results to be returned.
+* define a `string` field `next_page_token` in the `List` method's response message. This field represents the pagination token to retrieve the next page of results. If the value is `""`, it means no further results for the request.
+To retrieve the next page of results, client **shall** pass the value of response's `next_page_token` in the subsequent `List` method call (in the request message's `page_token` field):
+
+```HTTP
+GET /v2/<name>/manifests/sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042/references/list?page_token=1&page_size=10&next_page_token=<token>
+```
+
+The above specifies that a tags response SHOULD be returned, from the start of the result set, ordered lexically, limiting the number of results to `n`. The response to such a request would look as follows:
```

Comment:
I don't believe there's any requirement for a particular ordering.  (I'd prefer that we leave an ordering requirement out so that implementing services can choose to return ordered results or not.)

Reply:
Would this be a problem for a standard notary client to get different ordered results from different registries?

Reply:
Can we say that without an `orderBy` parameter, the results MUST return by __ order?
Or, should we have a set of required and optional `orderBy` values? I know search is the Achilles heel to these discussions, but I don't know how we get around it. So, what can we do to have something minimal?